### PR TITLE
Adding tabbing UI + First batch of tabs

### PR DESCRIPTION
**Pull request by burzum619**
The tabs included in the PR are the following:

Getting Started
Step By Step
Formative Assessment
Things to consider

### CODE REVIEW

File: item-authoring-tool/src/app/toolsforteachers/crud/tabs/tft-step-by-step-tab.component/tft-step-by-step-tab.component.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export class TFTStepByStepTabComponent implements OnInit {
   }
 
   ngOnInit(): void {
-    if (!this.resource.stepByStep) {
+    if (!this.resource.stepByStep || this.resource.stepByStep.length === 0) {
```
Comment (millerts) [inline item-authoring-tool/src/app/toolsforteachers/crud/tabs/tft-step-by-step-tab.component/tft-step-by-step-tab.component.ts]: We should implement some sensible defaults on the backend.  I'll add this to my list tomorrow

**Comment by millerts** [comment]
@alex-ponce or @esotrope , can you weigh in on this review?  My frontend experience is a bit rusty.

File: item-authoring-tool/src/app/toolsforteachers/crud/tabs/tft-getting-started-tab.component/tft-getting-started-tab.component.html [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    </ckeditor>
+    <h3 class="mt-sm">Learning Goal</h3>
+    <ckeditor
+        formControlName="learningGoal"
+        (change)="onChange('learningGoal', $event)"
+        (afterPaste)="onChange('learningGoal', $event)"
+        debounce="1000">
+    </ckeditor>
+    <h3 class="mt-sm">Success Criteria</h3>
+    <ckeditor
```
Comment (millerts) [inline item-authoring-tool/src/app/toolsforteachers/crud/tabs/tft-getting-started-tab.component/tft-getting-started-tab.component.html]: Not sure where it is configured, but we'll want to make sure that only the features that are approved in the requirement are enabled.

Comment (millerts) [reply item-authoring-tool/src/app/toolsforteachers/crud/tabs/tft-getting-started-tab.component/tft-getting-started-tab.component.html]: This goes for any CKEditor call.

Comment (burzum619) [reply item-authoring-tool/src/app/toolsforteachers/crud/tabs/tft-getting-started-tab.component/tft-getting-started-tab.component.html]: It is configured here I believe. I'll update all ckeditors with the correct features before merging this PR.

File: item-authoring-tool/src/app/toolsforteachers/crud/tabs/models/tft-tabs.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
+export class Tab {
```
Comment (esotrope) [inline item-authoring-tool/src/app/toolsforteachers/crud/tabs/models/tft-tabs.ts]: i'd recommend interface > class

File: item-authoring-tool/src/app/toolsforteachers/crud/tabs/models/tft-tabs.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  id: string;
+  label: string;
+}
+
+export class Tabs {
+  static AllTabs: Tab[] = [
+    {
+      id: 'gettingStarted',
+      label: 'Getting Started'
```
Comment (esotrope) [inline item-authoring-tool/src/app/toolsforteachers/crud/tabs/models/tft-tabs.ts]: this feels like it belongs in a `.component` as `const tabs`

File: item-authoring-tool/src/app/toolsforteachers/crud/tabs/tft-formative-assessment-tab.component/tft-formative-assessment-tab.component.html [hunk 5]
```diff
@@ -0,0 +1,46 @@
+<form [formGroup]="form">
+  <div class="well-body">
+    <h3>Clarify</h3>
+    <ckeditor
+        formControlName="clarify"
+        (change)="onChange('clarify', $event)"
+        (afterPaste)="onChange('clarify', $event)"
+        debounce="1000">
+    </ckeditor>
+    <h3 class="mt-sm">Elicit</h3>
```
Comment (esotrope) [inline item-authoring-tool/src/app/toolsforteachers/crud/tabs/tft-formative-assessment-tab.component/tft-formative-assessment-tab.component.html]: i'd recommend vanilla css over these helpers.
```
<div class="wrapper">
  <h3>
  <ckeditor>
</div>

.wrapper + .wrapper { margin-top: ... }
```

File: item-authoring-tool/src/app/toolsforteachers/crud/tabs/tft-step-by-step-tab.component/tft-step-by-step-tab.component.ts [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    this.resource.stepByStep.splice(index, 1);
+  }
+
+  onMoveStepUp(index: number): void {
+    this.resource.stepByStep.splice(index - 1, 0,  this.resource.stepByStep.splice(index, 1)[0]);
+  }
+
+  onMoveStepDown(index: number): void {
+    this.resource.stepByStep.splice(index + 1, 0,  this.resource.stepByStep.splice(index, 1)[0]);
+  }
```
Comment (esotrope) [inline item-authoring-tool/src/app/toolsforteachers/crud/tabs/tft-step-by-step-tab.component/tft-step-by-step-tab.component.ts]: would need more context but in general I would recommend avoiding splice in favor of filter/slice or other methods that return a new array and do not have side-effects

File: item-authoring-tool/src/app/toolsforteachers/crud/tft-crud.component.ts [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+      chev.classList.remove('fa-chevron-left');
+      chev.classList.add('fa-chevron-right');
+    } else {
+      fly.classList.remove('col-md-12');
+      fly.classList.add('col-md-4');
+      panel.classList.remove('hidden');
+      chev.classList.remove('fa-chevron-right');
+      chev.classList.add('fa-chevron-left');
+    }
+  }
```
Comment (esotrope) [inline item-authoring-tool/src/app/toolsforteachers/crud/tft-crud.component.ts]: why cant we use `[ngClass]` for this stuff? manipulating the class list like this happens outside of the angular render cycle

**Pull request merged**